2019-10-26	Jingnong Qu	<andrewqu2000@ucla.edu>

* lib/timespec.h (timespec_cmp): Return 1 if a.tv_sec > b.tv_sec;
return -1 if a.tv_sec < b.tv_sec; return 0 otherwise.

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h~	2019-10-26 13:23:03.455313000 -0700
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec)
+    return 1;
+  else if (a.tv_sec < b.tv_sec)
+    return -1;
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
